cabal.project

```diff
@@ -29,199 +46,24 @@ constraints:
   , ListLike >= 4.7.3
   , witherable >= 0.4
 
-  -- Plutus dependency
-  , algebraic-graphs < 0.7
-
-package Win32-network
-  tests: True
-
-package io-classes
-  flags: +asserts
-
-package strict-stm
-  flags: +asserts
-
-package network-mux
-  flags: +asserts
-
-package ouroboros-network-testing
-  tests: True
-
-package mux
-  tests: True
-
-package ouroboros-network
-  tests: True
-  flags: +asserts +cddl
-
-
-package ouroboros-consensus
-  flags: +asserts
-
-package ouroboros-consensus-test
-  tests: True
-
-package ouroboros-consensus-mock
-  flags: +asserts
-
-package ouroboros-consensus-mock-test
-  tests: True
-
-package ouroboros-consensus-byron
-  flags: +asserts
-
-package ouroboros-consensus-byron-test
-  tests: True
-
-package ouroboros-consensus-shelley
-  flags: +asserts
-
-package ouroboros-consensus-shelley-test
-  tests: True
-
-package ouroboros-consensus-cardano
-  flags: +asserts
-
-package ouroboros-consensus-cardano-test
-  tests: True
-
-package ouroboros-consensus-cardano-tools
-  tests: True
-
-
-package cardano-binary
-  tests: False
-
-package cardano-crypto
-  tests: False
-
-package cardano-crypto-class
-  tests: False
-
-package cardano-crypto-wrapper
-  tests: False
-
-package cardano-prelude-test
-  tests: False
-
-package contra-tracer
-  tests: False
-
-package byron-spec-chain
-  tests: False
-
-package byron-spec-ledger
-  tests: False
-
-package cardano-ledger-byron
-  tests: False
-
-package cardano-ledger-shelley-ma-test
-  tests: False
-
-package cardano-ledger-byron-test
-  tests: False
-
-package goblins
-  tests: False
-
-package shelley-spec-ledger-test
-  tests: False
-
-package cardano-ledger-alonzo-test
-  tests: False
-
-package shelley-spec-non-integral
-  tests: False
-
-package small-steps
-  tests: False
-
-package small-steps-test
-  tests: False
-
-package plutus-tx
-  tests: False
-
-package plutus-core
-  tests: False
-
-package prettyprinter-configurable
-  tests: False
+tests: True
 
 allow-newer:
   size-based:template-haskell
 
 constraints:
+  -- CONSTRAINTS FOR UPSTREAM CARDANO PACKAGES
+  -- These should all be constraints in upstream packages, but are currently missing.
+  -- We should a) fix these upstream and b) revise the constraints  on the released
+  -- version in CHaP. Unfortunately revisions in CHaP don't work yet, revisit this
```

Comment:
> TODO Unfortunately revisions in CHaP don't work yet, revisit this when we fix it.

Reply:
Also added a link to https://github.com/input-output-hk/haskell.nix/issues/1675